### PR TITLE
Implemented TripAdvisor Integration (v.0.0.1) #5369 

### PR DESCRIPTION
**Pull request by HahaBill**
## Description

This PR illustrates the TripAdvisor integration with Mindsdb. 

3 out of 5 API references were implemented: 

Find Search: https://tripadvisor-content-api.readme.io/reference/searchforlocations
Location Details: https://tripadvisor-content-api.readme.io/reference/getlocationdetails
Location Reviews: https://tripadvisor-content-api.readme.io/reference/getlocationreviews

If this PR is a success then 2 other API references will be implemented smoothly and fast. 

Issue number: #5369 

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)

## Verification Process

To ensure the changes are working as expected:

 - [x]   Test Location: Specify the URL or path for testing.

API Reference: https://tripadvisor-content-api.readme.io/reference/overview
Unit tests: tests/handler_tests/test_tripadvisor_handler.py
Manual tests: mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_manual_test.py

 - [x]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

In README.md + JSON files in "json_response_examples"

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by HahaBill** [comment]
@MinuraPunchihewa @ZoranPandovski I made a new PR after sync fork as requested

File: mindsdb/integrations/handlers/tripadvisor_handler/json_response_examples/location_details.json [hunk 1]
```diff
@@ -0,0 +1,197 @@
+{
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/tripadvisor_handler/json_response_examples/location_details.json:1]: Hey @HahaBill,
I think we discussed that these JSON files can be removed.

File: mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_handler.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+    Attributes:
+        api_key (str): The unique API key to access Tripadvisor content.
+        api (TripAdvisorAPI): The `TripAdvisorAPI` object for checking the connection to the TripAdvisor API.
+    """
+
+    def __init__(self, name=None, **kwargs):
+        super().__init__(name)
+
+        print("INPUT: ", kwargs)
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_handler.py:42]: Can you remove all of the `print()` statements? If there is useful information to be logged, please use the `logger`.

File: mindsdb/integrations/handlers/tripadvisor_handler/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+requests
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/tripadvisor_handler/requirements.txt:1]: We do not need this as `requests` is part of the base requirements for MindsDB. If this is the only dependency for the handler, let's remove this file.

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/tripadvisor_handler/json_response_examples/location_details.json (hunk 1 above)
Comment (HahaBill) [reply mindsdb/integrations/handlers/tripadvisor_handler/json_response_examples/location_details.json:1]: Oh I thought I've already removed it. Will do.

File: mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_handler.py (hunk 2 above)
Comment (HahaBill) [reply mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_handler.py:42]: Alright got it.

File: mindsdb/integrations/handlers/tripadvisor_handler/requirements.txt (hunk 3 above)
Comment (HahaBill) [reply mindsdb/integrations/handlers/tripadvisor_handler/requirements.txt:1]: Also got it.

**Comment by HahaBill** [comment]
@MinuraPunchihewa Changes made!

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
@HahaBill All of the `print()` statements still seem to be in here?
There might be one other change we might need to make to the location of the tests. I will confirm this to you soon.

**Comment by HahaBill** [comment]
@MinuraPunchihewa Ahhh, I probably didn't press save in VS Code. All print statements should be deleted.

**Comment by HahaBill** [comment]
Sounds good, I'll keep an eye on that (regarding the tests).

File: mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_api.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            response: Response object with response data as application/json
+        """
+        url = url + "{locationId}/reviews?language={language}&key={api_key}&".format(
+            locationId=locationId, api_key=self.api_key, language=language
+        )
+
+        response = self.getResponse(url)
+        return response
+
+    def makeRequest(self, apiCall, **params):
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_api.py:146]: Can you please change the name to be consistent to other methods?

File: mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_table.py [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+            pandas.DataFrame: A pandas DataFrame containing the selected data.
+        """
+
+        tripAdvisor = self.handler.connect()
+
+        conditions = extract_comparison_conditions(query.where)
+
+        params = {}
+        filters = []
+        for op, arg1, arg2 in conditions:
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_table.py:24]: Can we simplify this as: 

```
allowed_keys = set(["searchQuery", "category", "phone", "address", "latLong", "radius", "radiusUnit", "language"])

for op, arg1, arg2 in conditions:
    if op == "or":
        raise NotImplementedError(f"OR is not supported")
    elif op == "=" and arg1 in allowed_keys:
        params[arg1] = arg2
    elif op != "=":
        raise NotImplementedError(f"Unknown op: {op}")
    else:
        filters.append([op, arg1, arg2])

```

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Thanks, it looks great. I've added just 2 minor changes

File: mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_api.py (hunk 4 above)
Comment (HahaBill) [reply mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_api.py:146]: Alright will do!

File: mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_table.py (hunk 5 above)
Comment (HahaBill) [reply mindsdb/integrations/handlers/tripadvisor_handler/tripadvisor_table.py:24]: Got it, will do for all the tables.

**Comment by HahaBill** [comment]
@ZoranPandovski Changes made!

**Review by ZoranPandovski — APPROVED** [review]
Thanks

**Pull request merged**